gazebo/rendering/RTShaderSystem.cc

```diff
@@ -717,3 +685,91 @@ double RTShaderSystem::ShadowSplitPadding() const
 {
   return this->dataPtr->shadowSplitPadding;
 }
+
+/////////////////////////////////////////////////
+void RTShaderSystem::UpdateShadows()
+{
+  this->dataPtr->updateShadows = true;
+}
+
+/////////////////////////////////////////////////
+void RTShaderSystem::UpdateShadows(ScenePtr _scene)
+{
+  if (!this->dataPtr->initialized)
+    return;
+
+  Ogre::SceneManager *sceneMgr = _scene->OgreSceneManager();
+
+  sceneMgr->setShadowTechnique(Ogre::SHADOWTYPE_TEXTURE_ADDITIVE_INTEGRATED);
+
+  // directional: assume there can only be one dir light and we always create
+  //              the shadow map for the dir light
+  // spot: update number of shadow textures based on number of shadow casting
+  //       spot lights
+  // point: not working yet
+  unsigned int dirLightCount = 1u;
+  unsigned int spotLightCount = 0u;
+  for (unsigned int i = 0; i < _scene->LightCount(); ++i)
+  {
+    LightPtr light = _scene->LightByIndex(i);
+
+    if (!light->CastShadows())
+      continue;
+
+    if (light->Type() == "spot")
+      spotLightCount++;
+  }
+
+  // 3 textures per directional light
+  sceneMgr->setShadowTextureCountPerLightType(Ogre::Light::LT_DIRECTIONAL, 3);
+
+  // spot light shadow count
+  sceneMgr->setShadowTextureCountPerLightType(Ogre::Light::LT_SPOTLIGHT, 1);
+
+  // \todo(anyone) make point light shadows work
+  sceneMgr->setShadowTextureCountPerLightType(Ogre::Light::LT_POINT, 0);
+
+  // \todo(anyone) include point light shadows when it is working
+  unsigned int dirShadowCount = 3 * dirLightCount;
+  unsigned int spotShadowCount = spotLightCount;
+  sceneMgr->setShadowTextureCount(dirShadowCount + spotShadowCount);
+
+  unsigned int texSize = this->dataPtr->shadowTextureSize;
+#if defined(__APPLE__)
+  // workaround a weird but on OSX if texture size at 2 and 3 splits are not
+  // halved
+  texSize = this->dataPtr->shadowTextureSize/2;
+#endif
+  sceneMgr->setShadowTextureConfig(0,
+      this->dataPtr->shadowTextureSize, this->dataPtr->shadowTextureSize,
+      Ogre::PF_FLOAT32_R, 1);
+  sceneMgr->setShadowTextureConfig(1, texSize, texSize, Ogre::PF_FLOAT32_R, 2);
+  sceneMgr->setShadowTextureConfig(2, texSize, texSize, Ogre::PF_FLOAT32_R, 3);
```

Comment:
I believe the last arguments have been added since this code was moved down here (`fsaa` parameter with values 1, 2, 3); just confirming that this is intentional

Reply:
thanks for catching this! that was for testing. I removed them. 7aa1d35.